torch/utils/data/dataloader.py

```diff
@@ -224,11 +226,14 @@ def __init__(self, dataset: Dataset[T_co], batch_size: Optional[int] = 1,
             # option. If this turns out to be useful in future, we can re-enable
             # this, and support custom samplers that specify the assignments to
             # specific workers.
-            if shuffle is not False:
+            if isinstance(dataset, IterDataPipe):
+                torch.utils.data.graph_settings.apply_shuffle_settings(dataset, shuffle=shuffle)
+            elif shuffle is not False:
```

Comment:
The branch that `shuffle` is not False will never be reached.
Could we add a return boolean value for `apply_shuffle_settings` function to check if `dataset` / data pipeline has at least one `Shuffler`?
If false, raise the following ValueError.

Reply:
Addressing both comments. It is not the first time when `isinstance(dp, IterDataPipe)` not doing the proper job. I think it is better if we untie IterDataPipe and IterDataset with something simple like:

```
class IterDataPipe(IterDataset):
  pass
```

Reply:
SGTM!!